ns1/resource_zone.go

```diff
@@ -284,6 +293,12 @@ func zoneCreate(d *schema.ResourceData, meta interface{}) error {
 	if _, err := client.Zones.Create(z); err != nil {
 		return err
 	}
+	if !d.Get("autogenerate_ns_record").(bool) {
```

Comment:
This always evaluates `true` as `autogenerate_ns_record` does not appear to be saved in state.  NS records will always be deleted.

Reply:
Should not be an issue anymore, I removed the `DiffSuppressFunc`